### ISSUE TITLE
bar / hist : gap between first bar and other bars with lw=0.0

### ISSUE THREAD

**Issue by damiendr** [matplotlib/matplotlib#4602]
I like to make solid bar plots without lines around each bar, eg:
```python
plt.hist(data, fc="gray", lw=0.0, alpha=0.5)
```

Problem: the first bar does not exactly line up with the other bars. There is a slight misalignment visible on the right and bottom edges. Depending on the window size this shows either as a white gap or a darker overlap (visible because alpha=0.5).

The other bars are all properly lined up.

Confirmed both in the notebook inline mode and the QT backend on OSX.
```matplotlib.__version__ = '1.4.3'```

Steps to reproduce:

```python
import matplotlib.pylab as plt
import numpy as np
plt.figure()
plt.hist(np.random.uniform(0.0, 1.0, size=1000), fc="gray", lw=0.0, alpha=0.5)
plt.show()
```
Then resize the window and observe.

**Comment by damiendr** [comment]
Examples: 
![gap](https://cloud.githubusercontent.com/assets/1659790/8569274/971aab56-256d-11e5-8ad9-e5c94e25c969.png)
![overlap](https://cloud.githubusercontent.com/assets/1659790/8569273/971a0142-256d-11e5-99ee-07c9f45610bc.png)

**Comment by tacaswell** [comment]
That is...interesting...

You might be interested in https://github.com/matplotlib/matplotlib/pull/4433 which I hope to get merged for the upcoming release but you can pull the relevant function out (There is a copy-pastable function for this at https://github.com/matplotlib/matplotlib/issues/643#issuecomment-63762568).

**Comment by tacaswell** [comment]
I bet this is related to #4584 .

**Comment by jenshnielsen** [comment]
I agree that it is related to #4584 I can reproduce the issue with Agg based backends whereas it looks different with GTK3Cairo and WX

**Comment by cimarronm** [comment]
This is the same issue as #4212. See that issue for more details on what is happening. @damiendr, you can use the following as a workaround:
```python
_, _, collections = plt.hist(np.random.uniform(0.0, 1.0, size=1000), fc="gray", lw=0.0, alpha=0.5)
collections[0].set_snap(True)
```

**Comment by efiring** [comment]
Closed by #6887
